src/styles/main.scss

```diff
@@ -0,0 +1,79 @@
+$width-day: 100px;
+$gap: 1px;
+$start-day: (
+  mon 0,
+  tue ($width-day) + $gap,
+  wed ($width-day * 2) + ($gap * 2),
+  thu ($width-day * 3) + ($gap * 3),
+  fri ($width-day * 4) + ($gap * 4),
+  sat ($width-day * 5) + ($gap * 5),
+  sun ($width-day * 6) + ($gap * 6)
+);
+
+body {
+  height: 100%;
+  margin: 0;
+}
+
+.main {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  height: 100vh;
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  width: ($width-day * 7) + ($gap * 6);
+  padding: 10px 0;
+  gap: $gap;
+
+  &__day {
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    box-sizing: border-box;
+    width: $width-day;
+    height: 100px;
+    background-color: #eee;
+    border: 1px solid black;
+    margin: 0;
+    padding: 0;
+    transition: transform 0.5s;
+
+    &:hover {
+      cursor: pointer;
+      background-color: #ffbfcb;
+      transform: translateY(-20px);
+    }
+  }
+
+  &__day:first-of-type {
+    margin-left: ($width-day * 6) + $gap * 6;
```

Comment:
The `margin-left` for `.calendar__day:first-of-type` is set to create an offset for the first day of the month, but it's using a fixed value based on the width of 6 days plus 6 gaps. This seems to be a mistake since the start day of the month should be dynamic based on the `$start-day` map. You should use the `$start-day` map to set the correct margin for the first day of the month.